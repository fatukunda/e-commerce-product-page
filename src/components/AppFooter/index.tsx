import React, { FunctionComponent } from "react";

const AppFooter: FunctionComponent = (): JSX.Element => {
  return (
    <footer className="m-4 text-center text-[.6rem] md:m-2 md:text-xs">
      Challenge by{" "}
      <a
        href="https://www.nu3.de/"
        target="_blank"
        rel="noreferrer"
        className="font-bold uppercase text-primary underline transition-all duration-300 hover:text-tertiary"
      >
        Nu3
      </a>
      . Coded by{" "}
      <a
        href="https://www.frankatukunda.com/"
        target="_blank"
        rel="noreferrer"
        className="font-bold uppercase text-primary underline transition-all duration-300 hover:text-tertiary"
      >
        Frank Atukunda
      </a>
    </footer>
  );
};

export default AppFooter;
