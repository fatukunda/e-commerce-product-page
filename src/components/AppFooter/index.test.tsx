import { render, cleanup, fireEvent } from "@testing-library/react";
import renderer from "react-test-renderer";
import { Provider as ReduxProvider } from "react-redux";

import AppFooter from ".";
import mockStore from "../mocks/store";

const renderAppFooter = () =>
  render(
    <ReduxProvider store={mockStore()}>
      <AppFooter />
    </ReduxProvider>
  );
describe("Tests for <AppFooter/>", () => {
  afterEach(cleanup);
  test("AppFooter: Snapshot", () => {
    const component = renderer.create(
      <ReduxProvider store={mockStore()}>
        <AppFooter />
      </ReduxProvider>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("Renders without crashing", () => {
    renderAppFooter();
  });
  it("Displays Heading links", () => {
    const { getByText } = renderAppFooter();
    expect(getByText("Frank Atukunda")).toBeInTheDocument();
  });
});
