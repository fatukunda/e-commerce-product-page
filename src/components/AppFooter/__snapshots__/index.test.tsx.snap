// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppHeader: Snapshot 1`] = `
<footer
  className="m-4 text-center text-[.6rem] md:m-2 md:text-xs"
>
  Challenge by
   
  <a
    className="font-bold uppercase text-primary underline transition-all duration-300 hover:text-tertiary"
    href="https://www.nu3.de/"
    rel="noreferrer"
    target="_blank"
  >
    Nu3
  </a>
  . Coded by
   
  <a
    className="font-bold uppercase text-primary underline transition-all duration-300 hover:text-tertiary"
    href="https://www.frankatukunda.com/"
    rel="noreferrer"
    target="_blank"
  >
    Frank Atukunda
  </a>
</footer>
`;

exports[`Tests for <AppFooter/> App: Snapshot 1`] = `
<footer
  className="m-4 text-center text-[.6rem] md:m-2 md:text-xs"
>
  Challenge by
   
  <a
    className="font-bold uppercase text-primary underline transition-all duration-300 hover:text-tertiary"
    href="https://www.nu3.de/"
    rel="noreferrer"
    target="_blank"
  >
    Nu3
  </a>
  . Coded by
   
  <a
    className="font-bold uppercase text-primary underline transition-all duration-300 hover:text-tertiary"
    href="https://www.frankatukunda.com/"
    rel="noreferrer"
    target="_blank"
  >
    Frank Atukunda
  </a>
</footer>
`;

exports[`Tests for <AppFooter/> AppFooter: Snapshot 1`] = `
<footer
  className="m-4 text-center text-[.6rem] md:m-2 md:text-xs"
>
  Challenge by
   
  <a
    className="font-bold uppercase text-primary underline transition-all duration-300 hover:text-tertiary"
    href="https://www.nu3.de/"
    rel="noreferrer"
    target="_blank"
  >
    Nu3
  </a>
  . Coded by
   
  <a
    className="font-bold uppercase text-primary underline transition-all duration-300 hover:text-tertiary"
    href="https://www.frankatukunda.com/"
    rel="noreferrer"
    target="_blank"
  >
    Frank Atukunda
  </a>
</footer>
`;

exports[`Tests for <AppFooter/> AppHeader: Snapshot 1`] = `
<footer
  className="m-4 text-center text-[.6rem] md:m-2 md:text-xs"
>
  Challenge by
   
  <a
    className="font-bold uppercase text-primary underline transition-all duration-300 hover:text-tertiary"
    href="https://www.nu3.de/"
    rel="noreferrer"
    target="_blank"
  >
    Nu3
  </a>
  . Coded by
   
  <a
    className="font-bold uppercase text-primary underline transition-all duration-300 hover:text-tertiary"
    href="https://www.frankatukunda.com/"
    rel="noreferrer"
    target="_blank"
  >
    Frank Atukunda
  </a>
</footer>
`;

exports[`Tests for <AppHeader/> AppHeader: Snapshot 1`] = `
<footer
  className="m-4 text-center text-[.6rem] md:m-2 md:text-xs"
>
  Challenge by
   
  <a
    className="font-bold uppercase text-primary underline transition-all duration-300 hover:text-tertiary"
    href="https://www.nu3.de/"
    rel="noreferrer"
    target="_blank"
  >
    Nu3
  </a>
  . Coded by
   
  <a
    className="font-bold uppercase text-primary underline transition-all duration-300 hover:text-tertiary"
    href="https://www.frankatukunda.com/"
    rel="noreferrer"
    target="_blank"
  >
    Frank Atukunda
  </a>
</footer>
`;
